packages/core/src/delegation/DelegationNode.ts

```diff
@@ -10,56 +10,197 @@
+    account: IDelegationNode['account'],
+    permissions: IDelegationNode['permissions']
+  ): DelegationNode {
+    return new DelegationNode({
+      id: UUID.generate(),
+      hierarchyId,
+      parentId,
+      account,
+      permissions,
+      childrenIds: [],
+      revoked: false,
+    })
+  }
+
+  /**
+   * Builds a new [DelegationNode] representing a root delegation node ready to be submitted to the chain for creation.
+   *
+   * @param account - The address of this delegation (and of the whole hierarchy under it).
+   * @param permissions - The set of permissions to associated with this delegation node.
+   * @param hierarchyDetails - The details associated with the delegation hierarchy (e.g. The CType hash of allowed attestations).
+   *
+   * @returns A new [DelegationNode] with a randomly-generated id.
+   */
+  public static newRoot(
+    account: IDelegationNode['account'],
+    permissions: IDelegationNode['permissions'],
+    hierarchyDetails: IDelegationHierarchyDetails
+  ): DelegationNode {
+    const nodeId = UUID.generate()
+
+    const newNode = new DelegationNode({
+      id: nodeId,
+      hierarchyId: nodeId,
+      account,
+      permissions,
+      childrenIds: [],
+      revoked: false,
+    })
+    newNode.hierarchyDetails = hierarchyDetails
+
+    return newNode
+  }
+
+  /**
+   * Lazily fetches the details of the hierarchy the node is part of and return its CType.
+   *
+   * @returns The CType hash associated with the delegation hierarchy.
+   */
+  public get cTypeHash(): Promise<string> {
+    return this.getHierarchyDetails().then((details) => details.cTypeHash)
+  }
+
+  /**
+   * [ASYNC] Fetches the details of the hierarchy this delegation node belongs to.
+   *
+   * @throws [[ERROR_HIERARCHY_QUERY]] when the hierarchy details could not be queried.
+   * @returns Promise containing the [[DelegationHierarchyDetails]] of this delegation node.
+   */
+  public async getHierarchyDetails(): Promise<IDelegationHierarchyDetails> {
+    if (!this.hierarchyDetails) {
+      const hierarchyDetails = await queryDetails(this.hierarchyId)
+      if (!hierarchyDetails) {
+        throw SDKErrors.ERROR_HIERARCHY_QUERY(this.hierarchyId)
+      }
+      this.hierarchyDetails = hierarchyDetails
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return this.hierarchyDetails!
+  }
+
+  /**
+   * [ASYNC] Fetches the parent node of this delegation node.
+   *
+   * @returns Promise containing the parent as [[DelegationNode]] or [null].
+   */
+  public async getParent(): Promise<DelegationNode | null> {
+    return this.parentId ? query(this.parentId) : Promise.resolve(null)
+  }
+
+  /**
+   * [ASYNC] Fetches the children nodes of this delegation node.
+   *
+   * If new nodes have been created with this node as parent, **it is advised to first sync up this node with the latest blockchain state by calling [DelegationNodeUtils.getSyncedState]**.
```

Comment:
To do this, something like a `.refetch()` or `.refresh()` method would be cool on the delegation node I think, returning a freshly queried version of the current node.

Reply:
Fixed in https://github.com/KILTprotocol/sdk-js/pull/406/commits/8a20ab281b2ba7d57cd26af3a0736060d6fa3309.